src/librustdoc/formats/cache.rs

```diff
@@ -27,8 +28,9 @@ use crate::visit_lib::RustdocEffectiveVisibilities;
 /// to be a fairly large and expensive structure to clone. Instead this adheres
 /// to `Send` so it may be stored in an `Arc` instance and shared among the various
 /// rendering threads.
-#[derive(Default)]
-pub(crate) struct Cache {
+pub(crate) struct Cache<'tcx> {
+    pub(crate) tcx: TyCtxt<'tcx>,
```

Comment:
Changing this is super-invasive. While it's possible that `Cache` should hold a `TyCtxt`, that should be changed in it's own PR (or at least commit), as it has implications for all of rustdoc, and should be discussed with the other maintainers. Doing it in the same commit makes this super-tricky to review.